bridges/otelslog/handler.go

```diff
@@ -140,13 +181,258 @@ func (h *Handler) Enabled(ctx context.Context, l slog.Level) bool {
+	out := log.Map(g.name, g.attrs.KeyValues(kvs...)...)
+	g = g.next
+	for g != nil {
+		// A Handler should not output groups if there are no attributes.
+		if g.attrs.Len() > 0 {
+			out = log.Map(g.name, g.attrs.KeyValues(out)...)
+		}
+		g = g.next
+	}
+	return out
+}
+
+// Clone returns a copy of g.
+func (g *group) Clone() *group {
+	if g == nil {
+		return g
+	}
+	g2 := *g
+	g2.attrs = g2.attrs.Clone()
+	return &g2
+}
+
+// AddAttrs add attrs to g.
+func (g *group) AddAttrs(attrs []slog.Attr) {
+	if g.attrs == nil {
+		g.attrs = newKVBuffer(len(attrs))
+	}
+	g.attrs.AddAttrs(attrs)
+}
+
+var kvBufferPool = sync.Pool{
+	New: func() any {
+		// Based on slog research (https://go.dev/blog/slog#performance), 95%
+		// of use-cases will use 5 or less attributes.
+		return newKVBuffer(5)
+	},
+}
+
+func getKVBuffer() (buf *kvBuffer, free func()) {
+	buf = kvBufferPool.Get().(*kvBuffer)
+	return buf, func() {
+		// TODO: limit returned size so the pool doesn't hold on to very large
+		// buffers. Idea is based on
+		// https://cs.opensource.google/go/x/exp/+/814bf88c:slog/internal/buffer/buffer.go;l=27-34
+
+		// Do not modify any previously held data.
+		buf.data = buf.data[:0:0]
+		kvBufferPool.Put(buf)
+	}
+}
+
+type kvBuffer struct {
+	data []log.KeyValue
+}
+
+func newKVBuffer(n int) *kvBuffer {
+	return &kvBuffer{data: make([]log.KeyValue, 0, n)}
+}
+
+// Len returns the number of [log.KeyValue] held by b.
+func (b *kvBuffer) Len() int {
+	if b == nil {
+		return 0
+	}
+	return len(b.data)
+}
+
+// Clone returns a copy of b.
+func (b *kvBuffer) Clone() *kvBuffer {
+	if b == nil {
+		return nil
+	}
+	return &kvBuffer{data: slices.Clone(b.data)}
+}
+
+// KeyValues returns kvs appended to the [log.KeyValue] held by b.
+func (b *kvBuffer) KeyValues(kvs ...log.KeyValue) []log.KeyValue {
+	if b == nil {
+		return kvs
+	}
+	return append(b.data, kvs...)
+}
+
+// AddAttrs adds attrs to b.
+func (b *kvBuffer) AddAttrs(attrs []slog.Attr) {
+	b.data = slices.Grow(b.data, len(attrs))
+	for _, a := range attrs {
+		_ = b.AddAttr(a)
+	}
+}
+
+// AddAttr adds attr to b and returns true.
+//
+// This is designed to be passed to the AddAttributes method of an
+// [slog.Record].
+//
+// If attr is a group with an empty key, its values will be flattened.
+//
+// If attr is empty, it will be dropped.
+func (b *kvBuffer) AddAttr(attr slog.Attr) bool {
+	if attr.Key == "" {
+		if attr.Value.Kind() == slog.KindGroup {
+			// A Handler should inline the Attrs of a group with an empty key.
+			for _, a := range attr.Value.Group() {
+				b.data = append(b.data, log.KeyValue{
+					Key:   a.Key,
+					Value: convertValue(a.Value),
+				})
+			}
+			return true
+		}
+
+		if attr.Value.Any() == nil {
+			// A Handler should ignore an empty Attr.
+			return true
+		}
+	}
+	b.data = append(b.data, log.KeyValue{
+		Key:   attr.Key,
+		Value: convertValue(attr.Value),
+	})
+	return true
+}
+
+func convertValue(v slog.Value) log.Value {
```

Comment:
Should we document (in Go doc so it is clear for the users) how the bridge slog types are converted to OTel log records?
Especially how `slog.KindAny`, `slog.KindDuration`, `slog.KindTime` are converted.

This can be addressed in a separate PR.